### DISCUSSION TITLE
Martingale

### DISCUSSION THREAD

**Discussion by kenorb** [EA31337/EA31337-strategies#28] [Strategy Ideas]
Refs:
- https://www.forexfactory.com/showthread.php?t=772074
- https://corporatefinanceinstitute.com/resources/capital-markets/martingale-strategy/

**Comment by mavidelisi** [comment]
Kenorb Hello. I'll add a grid martingale strategy. Can you show me the way? For example, the grid distance is 20 pips and martin lotkoef is 1.4. I checked the source code, but I couldn't get out of it. multiplying the lot amount by 1.4 at certain distances, the process will continue until you get take profit. How can I do it, can you help me ? where do I need to change the source code or where do I need to add code? Thank you

**Reply by kenorb** [reply]
Strategy instance got access to Trade instance which have list of the current trades (which you can read via `GetOrdersActive()`. You can traverse list of active trades to find the right distance by querying each individual position of the strategy (for code example check `OrdersLoadByMagic()`). Where to add the code, is in `SignalOpen()` which should receive trade signal. Other methods such as `OnOrderOpen()` or `OnPeriod()` can be useful. If needed, you can define new extra variables in the strategy's param struct.
If this is not clear enough, I can try to write something similar.
Although my version of Martingale would not have a fixed grid (a standard implementation), but dynamic one, e.g. open sequential trades based on indicator signals.

**Reply by mavidelisi** [reply]
Thanks. İ will check

**Reply by mavidelisi** [reply]
i'm making a fixed grid, but if you have a ready-made dynamic grid, I'd like to see.
Thank you

**Comment by kenorb** [comment]
I've created two martingale-like strategies: [Oscillator Martingale](https://github.com/EA31337/Strategy-Oscillator_Martingale) and [Meta Martingale](https://github.com/EA31337/Strategy-Meta_Martingale). The first one uses oscillator signals, the second one another's strategy signals. Both are calculating min and max prices of the active orders (on top of the existing signals) and raising additional dynamic signals when price is beyond the existing price levels (adding extra orders, despite the original signal). Hope that helps. Currently it's working on fixed lot sizes, however increasing lot sizes based on some criteria can be added in the future versions, so it can increase probability of recovering the losing trades.

**Reply by mavidelisi** [reply]
hello, kenorb. I tested both strategies on eurusd and gbpusd, but I could not open an order in any way, I also changed the parameters, but still I was not successful. Is there a bug in the code? Have you tested the code? If you have tested it, is it possible for me to get the setting file? I'm still analyzing the codes, I'm planning to add dynamic and auto lot parameters and use them in the live account. i'm looking forward to version 3. i hope your studies are going well? thank you.

**Reply by kenorb** [reply]
Not sure why it would not open trades, I think it did for me last time when I've checked. I'll check on the next occasion.

**Reply by mavidelisi** [reply]
Ok. Thanks
